internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -2651,3 +2704,66 @@ func flattenKubernetesClusterMaintenanceConfigurationTimeInWeeks(input *[]contai
 	}
 	return results
 }
+
+func expandKubernetesClusterHttpProxyConfig(input []interface{}) *containerservice.ManagedClusterHTTPProxyConfig {
+	httpProxyConfig := containerservice.ManagedClusterHTTPProxyConfig{}
+	if len(input) == 0 {
+		return nil
+	}
+
+	config := input[0].(map[string]interface{})
+	httpProxyConfig.HTTPProxy = utils.String(config["http_proxy"].(string))
+	httpProxyConfig.HTTPSProxy = utils.String(config["https_proxy"].(string))
+	httpProxyConfig.TrustedCa = utils.String(config["trusted_ca"].(string))
+	noProxyRaw := config["no_proxy"].([]interface{})
+	httpProxyConfig.NoProxy = utils.ExpandStringSlice(noProxyRaw)
+	return &httpProxyConfig
+}
+
+func flattenKubernetesClusterHttpProxyConfig(props *containerservice.ManagedClusterProperties, flattenedDefaultNodePool *[]interface{}) []interface{} {
+	results := make(map[string]interface{})
+	httpProxyConfig := props.HTTPProxyConfig
+	if httpProxyConfig != nil {
+		results["http_proxy"] = httpProxyConfig.HTTPProxy
+		results["https_proxy"] = httpProxyConfig.HTTPSProxy
+		noProxyList := httpProxyConfig.NoProxy
+		// Removing the default values that AKS always applies from the flatten
+		newNoProxyList := make([]string, len(*noProxyList))
+		ignoreList := []string{
+			"168.63.129.16",
+			"169.254.169.254",
+			"127.0.0.1",
+			"localhost",
+			"konnectivity",
+		}
```

Comment:
Is this documented somehow?
```suggestion
		ignoreList := []string{
			"168.63.129.16",
			"169.254.169.254",
			"127.0.0.1",
			"localhost",
			"konnectivity",
		}
```